### DISCUSSION TITLE
Checking sensor status

### DISCUSSION THREAD

**Discussion by lurendrejer** [zwave-js/node-red-contrib-zwave-js#115] [Q&A] [answered]
Hi - how would i go about doing: Button pressed -> if door sensor closed -> then do whatever ?

I can't really see any way to check the status of a node in the 'plugin'

**Comment by marcus-j-davies** [comment, marked as answer]
Hey!

There are 2 ways.

Store the value of when the door was last closed/opened (storing the event somewhere) - for later use

OR.

use the `ValueAPI` - to fetch the value from the z-wave value cache


```javascript
let ValueID = {
    ....,
    ....
}
let Message = {
    payload: {
        mode: "ValueAPI",
        node: 2,
        method: "getValue",
        params: [ValueID]
    }
}
return Message
```

you can get the value ID, by double clicking the current value in the UI
remember this?
https://github.com/zwave-js/node-red-contrib-zwave-js/discussions/110#discussioncomment-1396951

**Reply by lurendrejer** [reply]
I'm going with the get-from cache part - I wan't things to be completely stateless :)

**Reply by marcus-j-davies** [reply]
The value cache is kept upto date as and when your devices send such updates.
so when you fetch the value as above - it will be the last value the device (door sensor) reported

some values are indeed stateless.
like a button press on a security keypad for instance.

**Reply by marcus-j-davies** [reply]
you could use the CCAPI also - but the CCAPI is 'real time' - so using a get type request wont work for battery devices.
the value cache will always be the last value that was known.

**Comment by marcus-j-davies** [comment]
here is the wiki for ValueAPI 😄 

https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/Value-API

**Comment by marcus-j-davies** [comment]
@lurendrejer 
Just a curtesy call, to see how things are going (and your still with us 😉 )

Any problems - please reach out to us

**Comment by lurendrejer** [comment]
Everything is dandy and I'm out for the weekend :)

I'm looking into node-red and clustering atm for a work-related case.
